packages/frontend-api/src/Model/Token/TokenFacade.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Shopsys\FrontendApiBundle\Model\Token;
+
+use Lcobucci\JWT\Builder;
+use Lcobucci\JWT\Parser;
+use Lcobucci\JWT\Signer;
+use Lcobucci\JWT\Signer\Key;
+use Lcobucci\JWT\Token;
+use Lcobucci\JWT\ValidationData;
+use Shopsys\FrameworkBundle\Component\Domain\Domain;
+use Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser;
+use Shopsys\FrontendApiBundle\Model\Token\Exception\ExpiredTokenUserMessageException;
+use Shopsys\FrontendApiBundle\Model\Token\Exception\InvalidTokenUserMessageException;
+use Shopsys\FrontendApiBundle\Model\Token\Exception\NotVerifiedTokenUserMessageException;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface;
+
+class TokenFacade
+{
+    protected const ACCESS_TOKEN_EXPIRATION = 3600;
+
+    /**
+     * @var \Shopsys\FrameworkBundle\Component\Domain\Domain
+     */
+    protected $domain;
+
+    /**
+     * @var \Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface
+     */
+    protected $parameterBag;
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Component\Domain\Domain $domain
+     * @param \Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface $parameterBag
+     */
+    public function __construct(
+        Domain $domain,
+        ParameterBagInterface $parameterBag
+    ) {
+        $this->domain = $domain;
+        $this->parameterBag = $parameterBag;
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser $customerUser
+     * @return string
+     */
+    public function generateAccessTokenByCustomerUser(CustomerUser $customerUser): string
+    {
+        return $this->generateTokenByCustomerUserAndExpiration($customerUser, static::ACCESS_TOKEN_EXPIRATION);
+    }
+
+    /**
+     * @param \Shopsys\FrameworkBundle\Model\Customer\User\CustomerUser $customerUser
+     * @param int $expiration
+     * @return string
+     */
+    protected function generateTokenByCustomerUserAndExpiration(CustomerUser $customerUser, int $expiration): string
+    {
+        $currentTime = time();
+
+        $tokenBuilder = (new Builder())
+            ->issuedBy($this->domain->getUrl())
+            ->permittedFor($this->domain->getUrl())
+            ->issuedAt($currentTime)
+            ->expiresAt($currentTime + $expiration);
+
+        foreach (TokenCustomerUserTransformer::transform($customerUser) as $key => $value) {
+            $tokenBuilder->withClaim($key, $value);
+        }
+
+        return (string)$tokenBuilder->getToken($this->getSigner(), $this->getKey());
+    }
+
+    /**
+     * @return \Lcobucci\JWT\Signer\Key
+     */
+    public function getKey(): Key
+    {
+        return new Key($this->parameterBag->get('secret'));
```

Comment:
I still struggle with this, as we spoke earlier. 
Right now it's really to enable frontend-api without changing the default secret. That would be fatal for any production shop (considering it the same as leaving default admin with the default password in production, but much easier to do by mistake as it's not in the DB, but in parameters.yml file).
I think it's really important to keep the key in a different location than DB as it's now.

documentation about signing https://lcobucci-jwt.readthedocs.io/en/latest/configuration/

What do you say?

Reply:
We add US to sign tokens with a certificate. :+1: